packages/base/lua/Environment.lua

```diff
@@ -63,6 +63,39 @@ local function createRandomPlacement(environment, cs, max, placement)
 	end)
 end
 
+local function createSpreadPlacement(environment, cs, max, placement)
+	local nplacement = placement
+	if nplacement == nil then
+		nplacement = "placement"
+	end
+
+	local g = Group{}
+	forEachOrderedElement(environment, function(_, element)
+		local t = type(element)
+		if t == "Society" then
+			element.placements[nplacement] = cs
+			forEachAgent(element, function(agent)
+				g:add(agent)
+			end)
+		elseif t == "Agent" then
+			g:add(element)
+		end
+	end)
+
+	local traj = Trajectory{target = cs, select = function(cell)
+		return #cell:getAgents() < max
+	end}
+
+	g:randomize()
+	forEachAgent(g, function(agent)
+		local cell = traj:sample()
+		agent:enter(cell)
+		if #cell:getAgents() >= max then
+			traj = Trajectory{target = traj, select = traj.select}
```

Comment:
This takes too long to execute. Store the cells in a separate data structure and remove only one when it is full.